### PR TITLE
reworked release script

### PR DESCRIPTION
**Pull request by gdementen**

### CODE REVIEW

File: releaser/make_release.py [hunk 1]
```diff
@@ -354,10 +354,14 @@ def run_steps(config, steps, steps_funcs):
     for step_func, step_desc in steps_funcs[start:stop]:
         if step_desc:
             print(step_desc + '...', end=' ')
-            step_func(**config)
+
+        config_update = step_func(**config)
```
Comment (alixdamman) [inline releaser/make_release.py:380]: What is the interest of updating the `config` dict after the call to `step_func()`? There is one but I don't see it.

**Review by alixdamman — APPROVED** [review]
This is clearer like that. Thanks for the changes.

File: releaser/make_release.py (hunk 1 above)
Comment (gdementen) [reply releaser/make_release.py:380]: so that steps can use information provided/computed in a previous step and not only in set_config.

Comment (alixdamman) [reply releaser/make_release.py:380]: Ok. Thanks for the explanation.

**Pull request merged**